src/cryptonote_core/service_node_list.cpp

```diff
@@ -1534,21 +1552,22 @@ namespace service_nodes
     {
       std::lock_guard<boost::recursive_mutex> lock(m_sn_mutex);
 
-      for(const auto& kv_pair : m_transient_state.quorum_states)
-      {
-        quorum_for_serialization quorum = {};
-        quorum.version                  = get_min_service_node_info_version_for_hf(hf_version);
-        quorum.height                   = kv_pair.first;
-        quorum_manager const &manager   = kv_pair.second;
+      for (const auto &qs : {m_transient_state.old_quorum_states, m_transient_state.quorum_states})
```

Comment:
Does this create temporary copies of `quorum_states` and `old_quorum_states`?

Reply:
Yeah, you're right, that's not good code.  Will fix.

Reply:
Fixed in 32a17df